src/views/DataSessionsView.vue

```diff
@@ -23,41 +22,38 @@ onBeforeMount(()=>{
 onMounted(async () => {
 	await loadSessions()
 	// if user created or selected a specific datasession, load that tab
-	if (route.params.sessionId) {
+	if (route.params.sessionId && dataSessions.value.some(ds => ds.id == route.params.sessionId)) {
 		tab.value = Number(route.params.sessionId)
 		// if user is navigating to just /datasessions then their first datasession loads and adds /[first id] as params
 	} else {
 		if (dataSessions.value.length > 0) {
 			const firstSessionId = dataSessions.value[0].id
 			tab.value = firstSessionId
-			router.replace({ name: 'DataSessionDetails', params: { sessionId: firstSessionId }})
 		} else {
 			console.log('no data sessions available to display')
 		}
-		
 	}
 })
 
-function onTabChange(newSessionId) {
-	router.push({ name: 'DataSessionDetails', params: { sessionId: newSessionId }})
+function changeTab(newSessionId) {
+	tab.value = newSessionId
 }
 
-async function deleteSession(id) {
+function deleteDialog(id) {
 	deleteSessionId.value = id
 	showDeleteDialog.value = true
 }
 
 async function loadSessions() {
-	await fetchApiCall({url: dataSessionsUrl, method: 'GET', successCallback: (data) => {dataSessions.value = data.results}, failCallback: handleError})
-}
-
-function selectTab(index) {
-	if (index == selectedTab.value) {
-		selectedTab.value = -1
-	}
-	else {
-		selectedTab.value = index
+	// if tab is not in new data default to displaying first tab
+	function updateData(data) {
```

Comment:
Is there a way in which this nested function can be its own separate successCallback?